openedx/adg/lms/applications/tests/test_views.py

```diff
@@ -416,232 +436,419 @@ def test_get_redirects_without_login_for_education_experience_view():
+    complete required data added from the previous step of the application i.e Contact Info page
     """
-    user.application_hub.delete()
-    response = logged_in_client.get(reverse('application_education_experience'))
-    assert reverse('application_hub') in response.url
+    mock_render = mocker.patch('openedx.adg.lms.applications.views.render')
+
+    user.application_hub.is_written_application_completed = is_written_app_completed
+    if has_user_application:
+        UserApplicationFactory(user=user)
+    user.extended_profile.saudi_national = is_saudi
+    user.extended_profile.birth_date = VALID_USER_BIRTH_DATE_FOR_APPLICATION if has_birthdate else None
+
+    request = request_factory.get(reverse('application_education_experience'))
+    request.user = user
+
+    response = EducationAndExperienceView.as_view()(request)
+
+    if will_redirect:
+        assert response.get('Location') == reverse('application_hub')
+        assert response.status_code == HTTP_302_FOUND
+    else:
+        mock_render.assert_called_with(request, 'adg/lms/applications/education_experience.html', mock.ANY)
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('is_written_app_completed, has_user_application, is_saudi, has_birthdate', [
+    (True, True, True, True),
+    (False, False, True, True),
+    (False, True, False, True),
+    (False, True, True, False),
+    (False, False, False, True),
+    (False, False, True, False),
+    (False, True, False, False)
+], ids=[
+    'written_app_completed',
+    'written_app_not_completed_with_no_application',
+    'written_app_not_completed_and_is_not_saudi',
+    'written_app_not_completed_and_has_no_birthdate',
+    'written_app_not_completed_with_no_application_and_is_not_saudi',
+    'written_app_not_completed_with_no_application_and_has_no_birthdate',
+    'written_app_not_completed_and_is_not_saudi_and_has_no_birthdate'
+])
+def test_post_without_required_prereqs_completed_education_experience_view(
+    is_written_app_completed, has_user_application, is_saudi, has_birthdate, logged_in_client, user
+):
+    """
+    Test that if a user tries to do a post request to the education_experience page with incomplete
+    prereqs e.g no education added, no work added etc, the user must be returned with status 400.
+    """
+    user.application_hub.is_written_application_completed = is_written_app_completed
+    if has_user_application:
+        UserApplicationFactory(user=user)
+    user.extended_profile.saudi_national = is_saudi
+    user.extended_profile.birth_date = VALID_USER_BIRTH_DATE_FOR_APPLICATION if has_birthdate else None
+
+    response = logged_in_client.post(reverse('application_education_experience'))
+    assert response.status_code == HTTP_400_BAD_REQUEST
 
 
 @pytest.mark.django_db
-def test_get_education_experience_view(request_factory, user):
+@pytest.mark.parametrize(
+    'button, background_question, template',
```

Comment:
the docstring states "...when all the required data is added". So why are we only testing for Background question here?

Reply:
Because the rest isn't really related to the view. It is more related to the api_view